analyzers/tests/SonarAnalyzer.Test/TestCases/AspNet/CallModelStateIsValid.cs

```diff
@@ -0,0 +1,225 @@
+    //            ^^^
+    {
+        return "Hello!";
+    }
+
+    [HttpPost("/[controller]")]
+    public string Post(Movie movie)                                             // Noncompliant
+    {
+        return "Hello!";
+    }
+
+    [HttpPut("/[controller]")]
+    public string Put(Movie movie)                                              // Noncompliant
+    {
+        return "Hello!";
+    }
+
+    [HttpDelete("/[controller]")]
+    public string Delete(Movie movie)                                           // Noncompliant
+    {
+        return "Hello!";
+    }
+
+    [HttpPatch("/[controller]")]
+    public string Patch(Movie movie)                                            // Noncompliant
+    {
+        return "Hello!";
+    }
+
+    [HttpGet]
+    [HttpPost]
+    [HttpPut]
+    [HttpDelete]
+    [HttpPatch]
+    [Route("/[controller]/mix")]
+    public string Mix([Required, FromQuery, EmailAddress] string email)         // Noncompliant
+    {
+        return "Hello!";
+    }
+
+    [AcceptVerbs("GET")]
+    [Route("/[controller]/accept-verbs")]
+    public string AGet([Required] string id)                                    // Noncompliant
+    {
+        return "Hello!";
+    }
+
+    [AcceptVerbs("POST")]
+    [Route("/[controller]/accept-verbs")]
+    public string APost(Movie movie)                                            // Noncompliant
+    {
+        return "Hello!";
+    }
+
+    [AcceptVerbs("PUT")]
+    [Route("/[controller]/accept-verbs")]
+    public string APut(Movie movie)                                             // Noncompliant
+    {
+        return "Hello!";
+    }
+
+    [AcceptVerbs("DELETE")]
+    [Route("/[controller]/accept-verbs")]
+    public string ADelete(Movie movie)                                          // Noncompliant
+    {
+        return "Hello!";
+    }
+
+    [AcceptVerbs("PATCH")]
+    [Route("/[controller]/accept-verbs")]
+    public string APatch(Movie movie)                                           // Noncompliant
+    {
+        return "Hello!";
+    }
+
+    [AcceptVerbs("GET", "POST", "PUT", "DELETE", "PATCH")]
+    [Route("/[controller]/many")]
+    public string Many([Required, FromQuery, EmailAddress] string email)        // Noncompliant
+    {
+        return "Hello!";
+    }
+
+    [HttpGet("/[controller]/list")]
+    public string[] List() => null;                                             // Compliant
+}
+
+[ApiController]
+public class ControllerWithApiAttributeAtTheClassLevel : ControllerBase
+{
+    [HttpPost("/[controller]")]
+    public string Add(Movie movie)                                              // Compliant, ApiController attribute is applied at the class level.
+    {
+        return "Hello!";
+    }
+}
+
+[Controller]
+public class ControllerThatDoesNotInheritFromControllerBase
+{
+    [HttpPost("/[controller]")]
+    public string Add(Movie movie)                                              // Compliant, ModelState is not available in this context.
```

Comment:
You may want to add a FN example with a FluentValidation dependency, where the validation is missing.